docs/cep/index.md

```diff
@@ -28,6 +28,90 @@ You can process streams to perform the following actions with your data:
+
+```sql
+CREATE TABLE GLOBAL UserTable (userId long, firstName string, lastName string);
+```
+
+#### Define the Sink Stream
+
+The sink stream is where the stream worker sends your data.
+
+```sql
+CREATE SINK EnrichedTransactionStream WITH (type='stream', stream='EnrichedTransactionStream', map.type='json') (userId long, userName string, transactionAmount double, location string);
+```
+
+### Data Enrichment Query
+
+Define the query for a stream to join the stream and the table, and then handle the result. This section examines the query line by line.
+
+#### Insert Data
+
+The `insert into` clause defines an output stream into which the enriched data is directed.
+
+```sql
+insert into EnrichedTransactionStream;
+```
+
+#### Select Data
+
+A `select` clause specifies how the value for each attribute in the output stream is derived. The variables used for the attributes are defined in the next line where you [join data](#join-data).
+
+```sql
+select u.userId, str:concat( u.firstName, " ", u.lastName) as userName, transactionAmount, location
+```
+
+Note the following in the `select` statement:
+
+- The `userId` attribute name is common to both the stream and the table. Therefore, you need to specify from where this attribute needs to be taken. Here, you can also specify `t.userId` instead of `u.userId`.
+- You are specifying the output generated to include an attribute named `userName`. The value for that is derived
+by concatenating the values of the `firstName` and `lastName` attributes in the `UserTable` table using the `str:concat()` function.
+- You can apply any of the range of streams functions available to further enrich the joined output.
+
+#### Join Data
+
+The `from` clause together with the `join` keyword join the table and the stream.
+
+```sql
+from UserTable as u join TransactionStream as t on u.userId == t.userId
+```
+
+Note the following about the `from` clause:
+
+- The input data is taken from both a stream and a table. You need to assign a unique reference for each of them to allow the query to differentiate between the common attributes. In this example, `TransactionStream` stream is referred to as `t`, and the `UserTable` table is referred to as `u`.
+- The `join` keyword joins the stream and the table together and defines the unique references.
+- The condition for the stream and the table to be joined is `t.userId == u.userId`, which means that for an event to be taken from the `TransactionStream` for the join, one or more events that have the same value for the `userId` must exist in the `UserTable` table and vice versa.
+
+
+
 ## Architecture
```

Comment:
We're going to need to seriously look at this Architecture and the Elements section at some point. If not for this ticket, then make new Jiras for them. Might end up making them separate topics, might just rewrite them. Not sure what makes the most sense.